### ISSUE TITLE
Bring where formatting inline with rusfmt

### ISSUE THREAD

**Issue by matklad** [intellij-rust/intellij-rust#2234]
Currently, rustfmt formats where clauses a bit differently that intellij rust:

```
fn foo<S>() 
where
    S: Clone,
{}
```

- `where` is not indented
- `where` is on its own line
- each where clause is on a newline as well

The relevant code is over here: https://github.com/intellij-rust/intellij-rust/tree/master/src/main/kotlin/org/rust/ide/formatter

It's important to add a setting (disabled by default) to preserve the old behavior.

**Comment by rcoh** [comment]
What about just adding the option for intellij to use rustfmt directly?

**Comment by Undin** [comment]
We already have the action to reformat code using `rustfmt` - "Reformat file with rustfmt"
But I think it's a good idea to have the option to use `rustfmt` as standard reformat action

**Comment by daboross** [comment]
Any chance of intellij-rust adding an option to change the behavior of the "Reformat" action too? Using "Reformat file with rustfmt" requires either using a different keybinding than "Reformat" or changing that keybinding and breaking formatting on all non-Rust files.

**Comment by Kobzol** [comment]
I agree that using rustfmt as a default reformat action could be a good idea. It's standardized in the Rust ecosystem and already well supported by IntelliJ. In the extreme case the Rust refactorings could be removed altogether to shed unnecessary code.

**Comment by matklad** [comment]
Note that even if `rustfmt` is used for code formatting, we still need internal formatter to build proper indent model to handle <kbd>Enter</kbd> and <kbd>Backspace</kbd> ([example](https://github.com/rust-analyzer/rust-analyzer/issues/4182#issuecomment-620500420)). In this sense, fixing indentation for where is still nedeed.
